### ISSUE TITLE
'yield' not allowed in an async function

### ISSUE THREAD

**Issue by htwenning** [microsoft/pyright#182]
**Describe the bug**
I use asynccontextmanager in project, but pyright exports <code>'yield' not allowed in an async function</code> error.

**To Reproduce**
write yield in an async funcion.

**Expected behavior**
no error exported.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```
from contextlib import asynccontextmanager

@asynccontextmanager
async def get_connection():
    conn = await acquire_db_connection()
    try:
        yield conn
    finally:
        await release_db_connection(conn)
```

**VS Code extension or command-line**

VS Code extension
Version: 1.0.52

pyright config:
```
{
  "include": ["./"],
  "exclude": ["./tmp", "./test", "./venv" ],
  "pythonVersion": "3.7",
  "venvPath": "../",
  "executionEnvironments": [
    {
      "root": "./",
      "venv": "venv"
    }
  ]
}
```

**Comment by erictraut** [comment]
Thanks for the bug report. I misinterpreted the statement in PEP 525 to mean that yield wasn't allowed in async statements, but upon closer reading, it says that "yield from" is not allowed while "yield" is. This bug will be fixed in the next version of pyright.

**Comment by erictraut** [comment]
This is fixed in 1.0.53, which I just published.
